test/functional/device/calendar-e2e-specs.js

```diff
@@ -1,40 +1,42 @@
 import chai from 'chai';
 import chaiAsPromised from 'chai-as-promised';
-import { TESTAPP_CAPS } from '../desired';
+import { TESTAPP_CAPS, amendCapabilities } from '../desired';
 import { initSession, deleteSession, MOCHA_TIMEOUT } from '../helpers/session';
 
 
 chai.should();
 chai.use(chaiAsPromised);
 
 if (!process.env.REAL_DEVICE && !process.env.CLOUD) {
-  describe('XCUITestDriver - calendar', function () {
+  describe('XCUITestDriver - calendar', function() {
     this.timeout(MOCHA_TIMEOUT);
 
     let driver, caps;
 
-    beforeEach(function () {
+    beforeEach(function() {
       caps = Object.assign({}, TESTAPP_CAPS);
     });
 
-    afterEach(async function () {
+    afterEach(async function() {
       await deleteSession();
     });
 
-    it('should authorize calendar access if calendarAccessAuthorized == true', async function () {
-      caps.calendarAccessAuthorized = true;
+    it('should authorize calendar access if calendarAccessAuthorized == true', async function() {
+      caps = amendCapabilities(caps, { 'appium:calendarAccessAuthorized': true });
       driver = await initSession(caps);
-      let checkCalendarButton = await driver.elementByName('Check calendar authorized');
+      let checkCalendarButton = await driver.$(`//XCUIElementTypeButton[@name='Check calendar authorized']`);
```

Comment:
prefeer to use predicates. xpath is slow

Reply:
Sure, fixed https://github.com/appium/appium-xcuitest-driver/pull/1461/commits/5c729dfdd71b6e4a4c3d5c400f67b70cda91c7da